server/src-lib/Hasura/Server/App.hs

```diff
@@ -288,17 +290,29 @@ mkSpockAction qErrEncoder qErrModifier serverCtx apiHandler = do
       json $ qErrEncoder includeInternal qErr
 
     logSuccessAndResp userInfo reqId req result qTime = do
-      logSuccess logger userInfo reqId req (apiRespToLBS result) qTime
+      let reqIdHeader = (requestIdHeader, unRequestId reqId)
```

Comment:
Does it make sense to make this a Wai Middleware and use it? So it is abstracted out, and to not introduce noise in business logic? Or am I over-thinking this?

Reply:
In `Middleware` implementation we should compress the `Response` and send it to the client. But the [library](http://hackage.haskell.org/package/wai-3.2.2.1/docs/Network-Wai.html#g:6) does not provide a function which simply does the following 
```haskell
modifyResponseBytes :: (ByteString -> ByteString) -> Response -> Response
```
We should pattern match on `Response` data type and handle all cases. That turns out to be doing more than what we required. Rather I think it's a library's job. I'm planning to stick with the current implementation.

Reply:
> How about adding a new function in this `where` block ...

The `logSuccess` requires the compressed response to log its size. Also, it requires the compression type. So, we need to compress beforehand, log and send it to client.